invoice/src/main/java/org/killbill/billing/invoice/InvoiceDispatcher.java

```diff
@@ -777,41 +780,56 @@ private boolean shouldIgnoreChildInvoice(final Invoice childInvoice, final BigDe
     public void processParentInvoiceForAdjustments(final ImmutableAccountData account, final UUID childInvoiceId, final InternalCallContext context) throws InvoiceApiException {
 
         final InvoiceModelDao childInvoiceModelDao = invoiceDao.getById(childInvoiceId, context);
-        final Invoice childInvoice = new DefaultInvoice(childInvoiceModelDao);
-        final InvoiceModelDao parentInvoice = childInvoiceModelDao.getParentInvoice();
+        final InvoiceModelDao parentInvoiceModelDao = childInvoiceModelDao.getParentInvoice();
 
-        if (parentInvoice == null) {
+        if (parentInvoiceModelDao == null) {
             throw new InvoiceApiException(ErrorCode.INVOICE_MISSING_PARENT_INVOICE, childInvoiceModelDao.getId());
-        } else if (parentInvoice.getStatus().equals(InvoiceStatus.COMMITTED)) {
-            // ignore parent invoice adjustment if it's in COMMITTED status.
-            return;
         }
 
+        final Long parentAccountRecordId = internalCallContextFactory.getRecordIdFromObject(account.getParentAccountId(), ObjectType.ACCOUNT, buildTenantContext(context));
+        final InternalCallContext parentContext = internalCallContextFactory.createInternalCallContext(parentAccountRecordId, context);
+        final String description = "Adjustment for account ".concat(account.getExternalKey());
+
         final InvoiceItemModelDao childInvoiceItemAdjustment = Iterables.find(childInvoiceModelDao.getInvoiceItems(), new Predicate<InvoiceItemModelDao>() {
             @Override
             public boolean apply(@Nullable final InvoiceItemModelDao input) {
                 return input.getType().equals(InvoiceItemType.ITEM_ADJ) || input.getType().equals(InvoiceItemType.REPAIR_ADJ);
```

Comment:
Thinking a bit more about it, we should probably pick the last created `ITEM_ADJ` on the child invoice. This is not a perfect solution as we could have scenarios where we have two concurrent `ITEM_ADJ` on the same child invoice.

A slightly better option would be to also also on the parent invoice to understand we we already had previous `ITEM_ADJ` and understand which one is missing. Code would become a bit more complex to write ( it think this is ok to implement the easier path for now as this is an edge case).

You should definitely enhance your previous test `testParentInvoiceItemAdjustmentUnpaidCommittedInvoice` to trigger several adjustments and make sure all of them are correctly accounted for.